fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/synthetic/JoinedRecordPlan.java

```diff
@@ -222,10 +222,14 @@ public JoinedRecordPlan(@Nonnull JoinedRecordType joinedRecordType, @Nonnull Lis
 
     private RecordCursor<EvaluationContext> query(int depth, @Nonnull FDBRecordStore store, @Nonnull EvaluationContext context,
                                                   @Nullable byte[] continuation, @Nonnull ExecuteProperties executeProperties) {
-        RecordCursor<FDBQueriedRecord<Message>> records = queries.get(depth).execute(store, context, continuation, executeProperties);
+        final RecordCursor<FDBQueriedRecord<Message>> records;
         final JoinedType joinedType = joinedTypes.get(depth + 1);
         if (joinedType.constituent.isOuterJoined()) {
-            records = records.orElse(executor -> RecordCursor.fromFuture(executor, CompletableFuture.completedFuture(null)));
+            records = RecordCursor.orElse(innerContinuation -> queries.get(depth).execute(store, context, innerContinuation, executeProperties),
```

Comment:
Given that the continuations for the cursors from the `JoinedRecordPlan` won't be valid any more, it seems like the `planHash` on this class should change (at least--perhaps ideally--if there are any outer joined types in `joinedTypes`).

Reply:
This class is package-private and minimally used so I don't think it's worth the additional code complexity.

Reply:
Yeah, package-private and we don't seem to be leaking the continuations from that planner, though maybe that's a sign that it shouldn't be `PlanHashable`, though not necessarily for this PR or anything. Okay, that's fine then.